Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Shared/BaseApiFixture.cs

```diff
@@ -0,0 +1,88 @@
+using AutoFixture;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+
+namespace Hackney.Core.Testing
+{
+    public abstract class BaseApiFixture<T> : IDisposable where T : class
```

Comment:
The `TenureApiFixture` class in the PersonListener repo has a more up-to-date version of the functionality for this class.